File: stripe/src/test/java/com/stripe/android/view/IconTextInputLayoutTest.java
package com.stripe.android.view;

import com.google.android.material.textfield.TextInputLayout;
import com.stripe.android.R;

import org.junit.After;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertTrue;

/**
 * Test class for {@link IconTextInputLayout} to ensure that the Reflection doesn't break
 * during an upgrade. This class exists only to wrap {@link TextInputLayout}, so there
 * is no need to otherwise test the behavior.
 */
@RunWith(RobolectricTestRunner.class)
public class IconTextInputLayoutTest extends BaseViewTest<CardInputTestActivity> {

    public IconTextInputLayoutTest() {
        super(CardInputTestActivity.class);
    }

    @After
    @Override
    public void tearDown() {
        super.tearDown();
    }

    @Test
    public void init_successfullyFindsFields() {
        final IconTextInputLayout iconTextInputLayout = createActivity()
                .getCardMultilineWidget()
                .findViewById(R.id.tl_add_source_card_number_ml);

        assertTrue(iconTextInputLayout.hasObtainedCollapsingTextHelper());
    }
}

File: stripe/src/test/java/com/stripe/android/view/IconTextInputLayoutTest.kt
package com.stripe.android.view

import android.content.Context
import androidx.test.core.app.ApplicationProvider
import com.google.android.material.textfield.TextInputLayout
import com.stripe.android.ApiKeyFixtures
import com.stripe.android.PaymentConfiguration
import com.stripe.android.R
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import kotlin.test.Test
import kotlin.test.assertTrue

/**
 * Test class for [IconTextInputLayout] to ensure that the Reflection doesn't break
 * during an upgrade. This class exists only to wrap [TextInputLayout], so there
 * is no need to otherwise test the behavior.
 */
@RunWith(RobolectricTestRunner::class)
internal class IconTextInputLayoutTest {

    private val context = ApplicationProvider.getApplicationContext<Context>()
    private val activityScenarioFactory = ActivityScenarioFactory(context)

    @Test
    fun init_successfullyFindsFields() {
        PaymentConfiguration.init(context, ApiKeyFixtures.FAKE_PUBLISHABLE_KEY)

        activityScenarioFactory
            .createAddPaymentMethodActivity()
            .use { activityScenario ->
                activityScenario.onActivity {
                    val iconTextInputLayout = it
                        .findViewById<CardMultilineWidget>(R.id.card_multiline_widget)
                        .findViewById<IconTextInputLayout>(R.id.tl_card_number)
                    assertTrue(iconTextInputLayout.hasObtainedCollapsingTextHelper())
                }
            }
    }
}
